code/datums/elements/objective_returner.dm

```diff
@@ -0,0 +1,100 @@
+/**
+ * # objective returner element!
+ *
+ * "The issue with steal objectives being off-station or destroyed could be solved by having
+ * a new one delivered/existing one retrieved by droppod to its owner after a random delay
+ * (to reduce gaming it like with the nuke disk). You'd have to do this with any object
+ * that could be an objective anytime (or any object that is an objective)." - caco
+ */
+/datum/element/objective_returner
+	element_flags = ELEMENT_BESPOKE|ELEMENT_DETACH
+	id_arg_index = 2
+	///the job datum of whatever job "owns" this steal objective. if none is given, it will be returned on the cargo shuttle
+	var/datum/job/job_datum_owner
+
+/datum/element/objective_returner/Attach(datum/target, job_datum_owner)
+	. = ..()
+	if(!ismovable(target))
+		return ELEMENT_INCOMPATIBLE
+	src.job_datum_owner = job_datum_owner
+	RegisterSignal(target, COMSIG_MOVABLE_Z_CHANGED, .proc/on_z_level_changed)
+	RegisterSignal(target, COMSIG_ITEM_DROPPED, .proc/on_dropped)
+	RegisterSignal(target, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+
+/datum/element/objective_returner/Detach(datum/target)
+	. = ..()
+	UnregisterSignal(target, list(COMSIG_MOVABLE_Z_CHANGED, COMSIG_ITEM_DROPPED, COMSIG_PARENT_EXAMINE))
+
+///signal called by examining the target
+/datum/element/objective_returner/proc/on_examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+
+	examine_list += "<span class='notice'>[source] has some kind of automatic return tech installed. \
+					If it is lost in deep space, it will return to its owner.</span>"
+
+///signal called by dropping the target
+/datum/element/objective_returner/proc/on_dropped(atom/movable/target, mob/dropper)
+	SIGNAL_HANDLER
+
+	if(!is_station_level(target.z) || !is_centcom_level(target.z))
```

Comment:
I get the feeling the objective is going to be returned when the shuttle leaves.
https://github.com/tgstation/tgstation/blob/4e2a36fe7e56fad93495adf01c697f4b5de04e90/code/datums/components/stationloving.dm#L66-L83

Reply:
I'll test vigorously!

Reply:
I tried this out yesterday. The two bugs I noticed were that dropping the captain's medal on the station would cause it to disappear and that it moved to transit space. I think it needs something like sparks and the teleport sound effect so people know it disappearing isn't a bug.